platform/posix/ota_pal/utest/ota_pal_posix_utest.c

```diff
@@ -0,0 +1,757 @@
+
+    chmod( OTA_PAL_UTEST_FIRMWARE_FILE, S_IRWXU );
+    result = OTA_PAL_MAIN_ERR( otaPal_CreateFileForRx( &otaFile ) );
+    TEST_ASSERT_EQUAL( OtaPalSuccess, result );
+}
+
+/**
+ * @brief Test that otaPal_CreateFileForRx will return correct result code.
+ */
+void test_OTAPAL_CreateFileForRx_ValidFileHandle( void )
+{
+    OtaPalMainStatus_t result;
+
+    otaFile.pFilePath = ( uint8_t * ) OTA_PAL_UTEST_FIRMWARE_FILE;
+    result = OTA_PAL_MAIN_ERR( otaPal_CreateFileForRx( &otaFile ) );
+    TEST_ASSERT_EQUAL( OtaPalSuccess, result );
+}
+
+
+/**
+ * @brief Test that otaPal_WriteBlock will return correct result code.
+ */
+void test_OTAPAL_WriteBlock_NullFileContext( void )
+{
+    int16_t bytes_written = 0;
+    uint8_t data = 0xAA;
+
+    bytes_written = otaPal_WriteBlock( NULL, 0, &data, 1 );
+    TEST_ASSERT_EQUAL( OtaPalSuccess, bytes_written + 1 );
+}
+
+
+/**
+ * @brief Test that otaPal_WriteBlock will return correct result code.
+ */
+void test_OTAPAL_WriteBlock_WriteSingleByte( void )
+{
+    OtaPalMainStatus_t result;
+    int16_t numBytesWritten;
+    uint8_t data = 0xAA;
+
+    /* TEST: Write a byte of data. */
+    otaFile.pFilePath = ( uint8_t * ) OTA_PAL_UTEST_FIRMWARE_FILE;
+    result = OTA_PAL_MAIN_ERR( otaPal_CreateFileForRx( &otaFile ) );
+    TEST_ASSERT_EQUAL( OtaPalSuccess, result );
+
+    if( TEST_PROTECT() )
+    {
+        numBytesWritten = otaPal_WriteBlock( &otaFile, 0, &data, 1 );
+        TEST_ASSERT_EQUAL_INT( 1, numBytesWritten );
+    }
+}
+
+/**
+ * @brief Test that otaPal_WriteBlock will return correct result code.
+ */
+void test_OTAPAL_WriteBlock_WriteManyBlocks( void )
+{
+    OtaPalMainStatus_t result;
+    int16_t numBytesWritten;
+
+    otaFile.pFilePath = ( uint8_t * ) OTA_PAL_UTEST_FIRMWARE_FILE;
+    otaFile.fileSize = sizeof( dummyData ) * testotapalNUM_WRITE_BLOCKS;
+    /* TEST: Write many bytes of data. */
+
+    otaFile.pFilePath = ( uint8_t * ) OTA_PAL_UTEST_FIRMWARE_FILE;
+    result = OTA_PAL_MAIN_ERR( otaPal_CreateFileForRx( &otaFile ) );
+    TEST_ASSERT_EQUAL( OtaPalSuccess, result );
+
+    if( TEST_PROTECT() )
+    {
+        int index = 0;
+
+        for( index = 0; index < testotapalNUM_WRITE_BLOCKS; index++ )
+        {
+            numBytesWritten = otaPal_WriteBlock( &otaFile, index * sizeof( dummyData ), dummyData, sizeof( dummyData ) );
+            TEST_ASSERT_EQUAL_INT( sizeof( dummyData ), numBytesWritten );
+        }
+
+        /* Sufficient delay for flash write to complete. */
+        /* vTaskDelay( pdMS_TO_TICKS( testotapalWRITE_BLOCKS_DELAY_MS ) ); */
+    }
+}
+
+/**
+ * @brief Test that otaPal_WriteBlock will return correct result code.
+ */
+void test_OTAPAL_WriteBlock_FseekError( void )
+{
+}
+
+/**
+ * @brief Test that otaPal_WriteBlock will return correct result code.
+ */
+void test_OTAPAL_WriteBlock_FwriteError( void )
+{
+}
+
+void test_OTAPAL_CloseFile_ValidSignature( void )
+{
+    OtaPalMainStatus_t result;
+    int16_t bytes_written = 0;
```

Comment:
Can this be implemented?  This seems like an important test to have.

Reply:
Will be addressed in further PR from @yanjos-dev